server/models/EarnedAchievement.coffee

```diff
@@ -118,9 +130,20 @@ EarnedAchievementSchema.statics.createForAchievement = co.wrap (achievement, doc
 
   if earnedDoc
     update = {$inc: {points: earnedPoints, 'earned.gems': earnedGems}}
+    user = yield User.findOne({_id: mongoose.Types.ObjectId(userID)})
     for rewardType, rewards of achievement.get('rewards') ? {}
-      continue if rewardType is 'gems'
-      if rewards.length
+      continue if rewardType is 'gems' or rewardType is 'heroes'
+      if rewardType is 'items'
+        # Unlock items if they aren't marked 'subscriber' true
+        earn = []
+        for reward in rewards
+          item = yield ThangType.findOne({ original: mongoose.Types.ObjectId(reward) }, { slug: 1, description: 1, subscriber: 1 });
+          if (not item.get('subscriber')) or (user.isPremium())
+            earn.push reward
+        if earn.length
+          update.$addToSet ?= {}
+          update.$addToSet["earned.#{rewardType}"] = { $each: earn }
+      else if rewards.length
```

Comment:
All logic in this file should have tests. Currently, the logic for `POST /db/earned_achievement` is in [achievement.spec.coffee](https://github.com/codecombat/codecombat/blob/vue-setup/spec/server/functional/achievement.spec.coffee#L209-209), that could be added to. I imagine tests would include:

* premium items are awarded if and only if the user is premium
* heroes are not rewarded

I would also refactor out this logic into a single function, since it's identical in both places.